### DISCUSSION TITLE
rbenv install 3.3.3 | mac OS 14.5 | M1 Chip | configure: exit 77

### DISCUSSION THREAD

**Discussion by j188482367** [rbenv/ruby-build#2420] [Build failures]
BUILD FAILED (macOS 14.5 on x86_64 using ruby-build 20240702)

You can inspect the build directory at /var/folders/ms/81b1597x1gnbqxncn1zlrhv40000gn/T/ruby-build.20240705154654.75014.bVSHVN
See the full build log at /var/folders/ms/81b1597x1gnbqxncn1zlrhv40000gn/T/ruby-build.20240705154654.75014.log

cd /var/folders/ms/81b1597x1gnbqxncn1zlrhv40000gn/T/ruby-build.20240705154654.75014.bVSHVN
==> Downloading ruby-3.3.3.tar.gz...
-> curl -q -fL -o ruby-3.3.3.tar.gz https://cache.ruby-lang.org/pub/ruby/3.3/ruby-3.3.3.tar.gz
-> tar xzf ruby-3.3.3.tar.gz
cd /var/folders/ms/81b1597x1gnbqxncn1zlrhv40000gn/T/ruby-build.20240705154654.75014.bVSHVN/ruby-3.3.3
==> Installing ruby-3.3.3...
-> ./configure "--prefix=$HOME/.rbenv/versions/3.3.3" --enable-shared --with-gmp-dir=/usr/local/opt/gmp --with-ext=openssl,psych,+ --with-openssl-dir=/usr/local/opt/openssl@1.1
checking for ruby... /Users/hom/.rvm/rubies/ruby-2.6.3/bin/ruby
tool/config.guess already exists
tool/config.sub already exists
checking build system type... x86_64-apple-darwin23.5.0
checking host system type... x86_64-apple-darwin23.5.0
checking target system type... x86_64-apple-darwin23.5.0
checking for ld... ld
checking for ar... ar
checking for clang++... clang++
checking for nm... nm
checking for objcopy... no
checking for objdump... objdump
checking for ranlib... ranlib
checking for strip... strip
checking for gcc... clang
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... configure: error: in `/var/folders/ms/81b1597x1gnbqxncn1zlrhv40000gn/T/ruby-build.20240705154654.75014.bVSHVN/ruby-3.3.3':
configure: error: cannot run C compiled programs.
If you meant to cross compile, use `--host'.
See `config.log' for more details
external command failed with status 77

[config.log](https://github.com/user-attachments/files/16106774/config.log)

**Comment by j188482367** [comment]
% sudo rm -rf /Library/Developer/CommandLineTools
Password:
% sudo xcode-select --install
xcode-select: note: install requested for command line developer tools
 
 Reinstalled command line developer tools .Got same issue

**Comment by mislav** [comment]
Hi, the title of what you posted mentions that you have an M1 machine. However, ruby-build thinks it's running under `x86_64` architecture, and so does the `configure` process for ruby:
```
checking build system type... x86_64-apple-darwin23.5.0
checking host system type... x86_64-apple-darwin23.5.0
checking target system type... x86_64-apple-darwin23.5.0
```
So, something doesn't match here. Are you using hacks like `arch -x86_64` to use Rosetta to run your programs in an emulated mode rather than on native `arm64` architecture? See `uname -m` in the same terminal where you are starting ruby-build.

**Comment by j188482367** [comment]
Damn follow this Tutorial ，Success 
[https://www.moncefbelyamani.com/how-to-install-xcode-homebrew-git-rvm-ruby-on-mac/](url)

**Reply by monfresh** [reply]
I'm glad my tutorial helped you!

**Reply by Terkea** [reply]
the tutorial was removed, can anybody explain please how to fix this?

**Reply by monfresh** [reply]
Here is the correct link for my [step by step guide for installing Ruby on a Mac](https://www.moncefbelyamani.com/how-to-install-xcode-homebrew-git-rvm-ruby-on-mac/).
